crda/remotematcher.go

```diff
@@ -83,6 +84,44 @@ func NewMatcher(opt ...Option) (*Matcher, error) {
 	return &m, nil
 }
 
+func getUrlFromString(params string) (string, string, error) {
+	c := strings.Split(params, " ")
+	tempURl := strings.Split(c[0], "=")
+	url := tempURl[1] + tempURl[2]
+	tempEcosystem := strings.Split(c[1], "=")
+	ecosystem := tempEcosystem[1]
+	return url, ecosystem, nil
+}
+
+// NewMatcher returns a configured Matcher or reports an error.
+func NewMatchers(params string, opt ...Option) (*Matcher, error) {
+	m := Matcher{}
+	for _, f := range opt {
+		if err := f(&m); err != nil {
+			return nil, err
+		}
+	}
+
+	URL, ecosystem, err := getUrlFromString(params)
+	fmt.Println(ecosystem)
+	if err == nil {
+		m.url, err = url.Parse(URL)
+		if err != nil {
+			return nil, err
+		} else {
+			m.url, err = url.Parse(defaultURL)
+			if err != nil {
+				return nil, err
+			}
+		}
+	}
+	if m.client == nil {
+		m.client = http.DefaultClient
+	}
+
+	return &m, nil
```

Comment:
You shouldn't add a new Matcher constructor, all you need to do is adding a `WithParam` function which parses the given string and constructs the matcher.

Refer WithURL, WithClient implementation.

Reply:
Yes I have written a function getUrlFromString(params) .So you want me to call it from inside the constructor you have created?

Reply:
Here is my expectation.

`m := NewMatcher(WithParam(param))`

Reply:
Got it.